addons/binding/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/eep/A5_11/A5_11_04.java

```diff
@@ -0,0 +1,196 @@
+        ENERGY_METERING_VALUE,
+        NOT_USED
+    }
+
+    public enum EnergyUnit {
+        MILLIWATT,
+        WATT,
+        KILOWATT,
+        MEGAWATT,
+        WATTHOUR,
+        KILOWATTHOUR,
+        MEGAWATTHOUR,
+        GIGAWATTHOUR,
+        NOT_SUPPORTED
+    }
+
+    private static Logger logger = LoggerFactory.getLogger(A5_11_04.class);
+
+    public A5_11_04(ERP1Message packet) {
+        super(packet);
+    }
+
+    protected boolean isErrorState() {
+        byte db0 = getDB_0();
+
+        int state = (db0 >> 4) & 0x03;
+
+        if (state != 0) {
+            // TODO: display error state on thing
+            logger.warn("Received error {}: {}", state, Error.values()[state]);
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    protected ParameterMode getParameterMode() {
+        int pm = (getDB_0() >> 1) & 0x03;
+        return ParameterMode.values()[pm];
+    }
+
+    protected EnergyUnit getEnergyUnit() {
+        int unit = getDB_1();
+        if (unit < 8) {
+            return EnergyUnit.values()[unit];
+        }
+
+        return EnergyUnit.NOT_SUPPORTED;
+    }
+
+    protected State getLightingStatus() {
+        byte db0 = getDB_0();
+        boolean lightOn = getBit(db0, 0);
+
+        return lightOn ? OnOffType.ON : OnOffType.OFF;
+    }
+
+    protected State getDimmerStatus() {
+        if (getParameterMode() == ParameterMode.EIGHT_BIT_DIMMER_VALUE_AND_LAMP_OPERATING_HOURS) {
+            return new PercentType(getDB_3Value() * 100 / 255);
+        }
+        return UnDefType.UNDEF;
+    }
+
+    protected State getEnergyMeasurementData() {
+        if (getParameterMode() == ParameterMode.ENERGY_METERING_VALUE) {
+            EnergyUnit unit = getEnergyUnit();
+
+            float factor = 1;
+            switch (unit) {
+                case WATTHOUR:
+                    factor /= 1000;
+                    break;
+                case KILOWATTHOUR:
+                    factor = 1;
+                    break;
+                case GIGAWATTHOUR:
+                    factor *= 1000;
+                case MEGAWATTHOUR:
+                    factor *= 1000;
+                    break;
+                default:
+                    return UnDefType.UNDEF;
+            }
+
+            return new QuantityType<>(
+                    Long.parseLong(HexUtils.bytesToHex(new byte[] { getDB_3(), getDB_2() }), 16) * factor,
+                    SmartHomeUnits.KILOWATT_HOUR);
+        }
+
+        return UnDefType.UNDEF;
+    }
+
+    protected State getPowerMeasurementData() {
+        if (getParameterMode() == ParameterMode.ENERGY_METERING_VALUE) {
+            EnergyUnit unit = getEnergyUnit();
+
+            float factor = 1;
+            switch (unit) {
+                case MILLIWATT:
+                    factor /= 1000;
+                    break;
+                case WATT:
+                    factor = 1;
+                    break;
+                case KILOWATT:
```

Comment:
This looks like a bug. `KILOWATT` falls through to `MEGAWATT`. I think it should be the other way around. Like on line 121 with `GIGAWATTHOUR`.

Reply:
Fixed this in #5054 as I am fixing another factor value there.